dependency/actions/compile/entrypoint

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+set -eu
+set -o pipefail
+
+readonly PROGDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+function main() {
+  local version output_dir target bundler_dir
+
+  while [ "${#}" != 0 ]; do
+    case "${1}" in
+      --version)
+        version="${2}"
+        shift 2
+        ;;
+
+      --outputDir)
+        output_dir="${2}"
+        shift 2
+        ;;
+
+      --target)
+        target="${2}"
+        shift 2
+        ;;
+
+      "")
+        shift
+        ;;
+
+      *)
+        echo "unknown argument \"${1}\""
+        exit 1
+    esac
+  done
+
+  if [[ -z "${version:-}" ]]; then
+    echo "--version is required"
+    exit 1
+  fi
+
+  if [[ -z "${output_dir:-}" ]]; then
+    echo "--outputDir is required"
+    exit 1
+  fi
+
+  if [[ -z "${target:-}" ]]; then
+    echo "--target is required"
+    exit 1
+  fi
+
+  echo "Downloading bundler gem"
+
+  local gem_archive="${PROGDIR}/bundler.gem"
+  curl "https://rubygems.org/downloads/bundler-${version}.gem" \
```

Comment:
In general I like to use `curl --fail` (or `--fail-with-body` for newer versions).

Otherwise the script will continue on when `curl` failed to download something. That almost always results in a more complex error to debug later.